### PR TITLE
clsdk: get_code_hash

### PR DESCRIPTION
**Pull request by tbfleming**

### CODE REVIEW

File: libraries/eosiolib/contracts/include/eosio/action.hpp [hunk 1]
```diff
@@ -225,6 +230,35 @@ namespace eosio
     */
    inline bool is_account(name n) { return internal_use_do_not_use::is_account(n.value); }
 
+   struct code_hash
+   {
+      varuint32 struct_version = {};
+      uint64_t code_sequence = {};
+      checksum256 code_hash = {};
```
Comment (swatanabe) [inline libraries/eosiolib/contracts/include/eosio/action.hpp:237]: This compiles?

Comment (tbfleming) [reply libraries/eosiolib/contracts/include/eosio/action.hpp:237]: yes. c++20 added additional rules making non-movable, non-copyable objects sometimes returnable, including from constructors.

Comment (tbfleming) [reply libraries/eosiolib/contracts/include/eosio/action.hpp:237]: Oh, the same name... I'm not sure why it compiles.

Comment (swatanabe) [reply libraries/eosiolib/contracts/include/eosio/action.hpp:237]: Apparently its okay as long as you have no user-declared constructors.  Probably for C compatibility.

**Review by swatanabe — APPROVED** [review]

**Review by swatanabe — APPROVED** [review]

**Pull request merged**